app/src/org/odk/collect/android/widgets/IntentWidget.java

```diff
@@ -107,6 +92,34 @@ public void onClick(View v) {
         // finish complex layout
         addView(launchIntentButton);
         addView(mStringAnswer);
+        
+        //only auto advance if 1) we have no data 2) its quick 3) we weren't just cancelled
+        if(s == null && ic.isQuickAppearance() && !ic.getCancelled()){
+            performCallout();
+        } else if(ic.getCancelled()){
+            //reset the cancelled flag
+            ic.setCancelled(false);
```

Comment:
the fact that this is loaded/managed by this widget really worries me, since I think there might be paths where this response isn't handled on the way. 

Do we have any way of detecting a "cold" start and making sure the flag is set to false there? If not this might be the best plan, it's just a generally very dangerous pattern to keep state in an object with a lifecycle that's not consistent.